sztp-agent/pkg/secureagent/utils.go

```diff
@@ -159,29 +160,44 @@ func generateInputJSONContent() string {
 
 type publicKey struct {
 	Algorithm string
-	KeyData   string
+	Data      string
 	Comment   string
 }
 
 func readSSHHostKeyPublicFiles(pattern string) []publicKey {
 	results := []publicKey{}
 	files, err := filepath.Glob(pattern)
 	if err != nil {
-		log.Printf("[ERROR] Error getting ssh host public keys file list : %v", err)
+		log.Printf("[ERROR] Error getting ssh host public keys file list: %v", err)
 		return results
 	}
 	for _, f := range files {
 		// nolint:gosec
-		data, _ := os.ReadFile(f)
-		// TODO: consider switching to https://pkg.go.dev/golang.org/x/crypto/ssh#ParseAuthorizedKey
-		parts := strings.Fields(string(data))
-		// [type-name] [base64-encoded-ssh-public-key] [comment]
-		if len(parts) < 2 {
-			log.Printf("[ERROR] Error parsing pub key, should contain at least 2 parts with spaces : %v", f)
+		data, err := os.ReadFile(f)
+		if err != nil {
+			log.Printf("[ERROR] Error reading public key file %s: %v", f, err)
+			return results
+		}
+
+		key, _, _, _, err := ssh.ParseAuthorizedKey(data)
+		if err != nil {
+			log.Printf("[ERROR] Problem parsing public key file %s: %v\n"+
+				"Check the key file has the correct format", f, err.Error())
 			continue
 		}
-		// ignore comment for now
-		results = append(results, publicKey{Algorithm: parts[0], KeyData: parts[1]})
+
+		keyParts := strings.Fields(string(data))
```

Comment:
can we remove this line and use `key.Marshal` instead of parsing string `data` ?

Reply:
Do we need to marshal to the wire format necessarily here? We could just return a slice of `ssh.PublicKey` instead and can use the underlying methods directly when needed elsewhere. 

Additionally:
```go
cert, ok := key.(*ssh.Certificate)
```
Does the type `PublicKey` inherit the interface `Certificate` directly to support casting to it? I seem to get a false `ok` when attempting to do so unless I am doing something wrong.

Reply:
If returning slice of Public key works, go for it.
As long as we get same base64 encoded result as in the file showed as examples in tests

Reply:
I can do this, it just depends if you would still like to maintain the ssh comment data? I don't think it's wholly necessary personally.